lib/decompress/zstd_decompress_block.c

```diff
@@ -970,10 +970,8 @@ ZSTD_decodeSequence(seqState_t* seqState, const ZSTD_longOffset_e longOffsets)
         } else {
             U32 const ll0 = (llBase == 0);
             if (LIKELY((ofBits == 0))) {
-                if (LIKELY(!ll0))
-                    offset = seqState->prevOffset[0];
-                else {
-                    offset = seqState->prevOffset[1];
+                offset = seqState->prevOffset[ll0];
+                if (UNLIKELY(ll0)) {
```

Comment:
That seems incorrect.
If we take this branch, then we should have `offset = seqState->prevOffset[1];`.
This statement has disappeared in the new formulation.
Consequently, I would expect the decompression to fail (corrupted data) whenever this case happens.